samples/match/match_non_enum.jakt

```diff
@@ -0,0 +1,12 @@
+function main() {
+    let x = match 42 {
+        (21 * 2) => [1, 2, 3]
+        else => []
+    }
+    let y = match "foo" {
+        ("f" + "oo") => "bar"
```

Comment:
Currently the expression matcher doesn't bind anything, we might have to rethink the syntax, or use the type name as a prefix: e.g. `Tuple(a, _)` for tuples or `Foo(a: b)` for structs/classes
I think the second one is an okay compromise, but I haven't thought carefully about it

Reply:
Ah yes, the cases can be constants (which makes the compiler emit switch-cases, or random expressions, which makes it emit if-else chains

Reply:
oooh, what if they paralleled the source for creation:

```
match foo {
    Foo(a: b) => { ... }
    (a, b) => { ... } <-- since that's how we make tuples
    {a} => { ... }  <-- does the set include this value
    (a: b) => { ... } <-- does the dictionary include this key
    5 => { ... } <-- is it 5?
}
```